### DISCUSSION TITLE
Proxying through nginx still tries to send event to plausible instance

### DISCUSSION THREAD

**Discussion by gjuric** [plausible/analytics#1334] [Self-Hosted Support] [answered]
Hi, I've successfully set proxying of plausible.js scripts to the plausible instance through the site I am tracking. The problem is that calls to `/api/event` are still sent to the plausible instance itself which is blocked by uBlock origin.

Upon further investigation I've noticed that the plausible script that gets loaded through http://example.com/script.js at the end of the script has `encodeURIComponent(e.message)}}(window,"https://stats.example.com")`.

How do I get rid of that reference to the plausible instance from plausible.js when if it's being served through proxy?

**Comment by ukutaht** [comment, marked as answer]
What rule is blocking it in uBlock? If your subdomain is actually named `stats.yourdomain.com` you can just rename it to something that's not blocked by uBlock. uBlock blocks `stats.*` and `analtyics.*` by default.

**Reply by gjuric** [reply]
The problem is that api events are not being proxied through example.com/api/event through nginx to stats.example.com/api/event although the script on the page is loaded from example.com/script.js and proxied successfully through nginx to stats.example.com/js/plausible.js. The html snippet on the page is also changed as per the instructions.

To simplify: loading the script through nginx proxy works, but for some reason javascript is not trying to send events through the proxy as well but is trying to connect directly to the plausible instance and this ofcourse is being blocked because of the domain name.

**Reply by ukutaht** [reply]
I understand your problem. It's not working because we made some changes to support the proxy which haven't been released for self-hosted yet.

In my answer I went straight to how I would fix it in your shoes. The API call would work fine if your sudomain was named something other than `stats` or `analytics`

**Reply by gjuric** [reply]
Ah, I do understand why it's being blocked but wanted to fix it properly without divulging the hostname where analytics are being hosted. Do you have an ETA on when these changes could be released to the self-hosted version?

Thank you for your work!

**Reply by ukutaht** [reply]
Ah yeah, `1.4.0` is scheduled for October or November. We release twice a year.
